libs/trucking-kit/company/data-access/src/lib/company.store.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable, computed, inject, signal } from '@angular/core';
+import { CompanyState, companyInitialState } from './company.model';
+import { Company } from '@shared/models';
+import { CompanyService } from './services/company.service';
+
+@Injectable({ providedIn: 'root' })
+export class CompanyStore {
+  protected state$ = signal<CompanyState>(companyInitialState);
+  protected service = inject(CompanyService);
+
+  $companies = computed(() => this.state$().companies);
+  $company = computed(() => this.state$().selected);
+  $creating = computed(() => this.state$().creating);
+  $fetching = computed(() => this.state$().fetching);
+  $loading = computed(() => this.state$().loading);
+  $removing = computed(() => this.state$().removing);
+  $updating = computed(() => this.state$().updating);
+
+  create(company: Company) {
+    this.state$.update((s) => ({ ...s, creating: true }));
+    this.service.createCompany(company).subscribe({
+      next: (result) =>
+        this.state$.update((s) => ({
+          ...s,
+          creating: false,
+          companies: s.companies.concat(result.data.createCompany),
+        })),
+      error: this.errorHandler((s) => ({ ...s, creating: false })),
+    });
+  }
+
+  delete(dotNumber: string) {
+    this.state$.update((s) => ({ ...s, removing: dotNumber }));
+    this.service.deleteCompany(dotNumber).subscribe({
+      next: () =>
+        this.state$.update((s) => ({
+          ...s,
+          removing: undefined,
+          companies: s.companies.filter((c) => c.dotNumber !== dotNumber),
+        })),
+      error: this.errorHandler((s) => ({ ...s, removing: undefined })),
+    });
+  }
+
+  fetchAll() {
+    this.state$.update((s) => ({ ...s, fetching: true }));
+    this.service.fetchCompanies().subscribe({
+      next: (res) =>
+        this.state$.update((s) => ({
+          ...s,
+          fetching: false,
+          companies: res.data.listCompanies.items,
+        })),
+      error: this.errorHandler((s) => ({ ...s, fetching: false })),
+    });
+  }
+
+  get(dotNumber: string) {
+    this.state$.update((s) => ({ ...s, loading: true, selected: undefined }));
+    this.service.getCompany(dotNumber).subscribe({
+      next: (res) =>
+        this.state$.update((s) => ({
+          ...s,
+          loading: false,
+          selected: res.data.getCompany,
+        })),
+      error: this.errorHandler((s) => ({ ...s, loading: false })),
+    });
+  }
+
+  update(changes: Partial<Company> & Required<Pick<Company, 'dotNumber'>>) {
```

Comment:
You may have had good success with the pattern of "selecting" a thing, and then that's what you are updating/working on. Or we could pass in the ID of the thing to update? Either way is fine, if you've seen this in action and think it has a good flow, then let's go with it.

Reply:
I may need some clarification on your meaning here. I assumed the store would only need to receive the Company properties that were updated, and could just bind the changes with the Company on the state. The only property required to provide is the dotNumber, for ensuring the api has the identifier.

Reply:
As in, select a company, and then you can update it. I don't really like having a whole "selected" company on the state. Rather just update whatever company is passed in the function, not the 'selected' one on the state.

So starting here, doing some selection and then updating is the part I don't really jive with.
`const company = this.state$().selected ?? ({} as Company);`